gptel-openai-extras.el

```diff
@@ -363,5 +363,71 @@ The Deepseek API requires strictly alternating roles (user/assistant) in message
     (setf (alist-get name gptel--known-backends nil nil #'equal) backend)
     backend))
 
+;;; xAI
+(cl-defstruct (gptel-xai (:include gptel-openai)
+                         (:copier nil)
+                         (:constructor gptel--make-xai)))
+
+;;;###autoload
+(cl-defun gptel-make-xai
+    (name &key curl-args stream key request-params
+          (header (lambda () (when-let* ((key (gptel--get-api-key)))
+                               `(("Authorization" . ,(concat "Bearer " key))))))
+          (host "api.x.ai")
+          (protocol "https")
+          (endpoint "/v1/chat/completions")
+          (models '((grok-3-latest
+                     :description "Grok 3"
+                     :capabilities '(text tool-use json)
```

Comment:
The `text` capability is not recognized -- it is assumed that all models gptel supports output text.  Please remove text from the `:capabilities` lists.